pkg/rhcos/builds.go

```diff
@@ -104,5 +108,14 @@ func fetchLatestBuild(ctx context.Context, channel string) (string, error) {
 		return "", errors.Errorf("no builds found")
 	}
 
+	if buildName != "" {
+		for _, build := range builds.Builds {
+			if build == buildName {
+				return buildName, nil
+			}
+		}
+		return "", errors.Errorf("no build with name '%s' found at '%s'. Available builds are: %v", buildName, url, builds.Builds)
```

Comment:
If we have a hard-coded `buildName`, I'd rather not bother hitting `builds.json` at all.  How about something like:

```diff
diff --git a/pkg/rhcos/ami.go b/pkg/rhcos/ami.go
index 2726e4c..1bfe041 100644
--- a/pkg/rhcos/ami.go
+++ b/pkg/rhcos/ami.go
@@ -8,7 +8,7 @@ import (
 
 // AMI fetches the HVM AMI ID of the latest Red Hat CoreOS release.
 func AMI(ctx context.Context, channel, region string) (string, error) {
-	meta, err := fetchLatestMetadata(ctx, channel)
+	meta, err := fetchMetadata(ctx, channel, buildName)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to fetch RHCOS metadata")
 	}
diff --git a/pkg/rhcos/builds.go b/pkg/rhcos/builds.go
index a1a7615..7ae32d0 100644
--- a/pkg/rhcos/builds.go
+++ b/pkg/rhcos/builds.go
@@ -32,10 +32,13 @@ type metadata struct {
 	OSTreeVersion string `json:"ostree-version"`
 }
 
-func fetchLatestMetadata(ctx context.Context, channel string) (metadata, error) {
-	build, err := fetchLatestBuild(ctx, channel)
-	if err != nil {
-		return metadata{}, errors.Wrap(err, "failed to fetch latest build")
+func fetchMetadata(ctx context.Context, channel string, build string) (metadata, error) {
+	var err error
+	if build == "" {
+		build, err = fetchLatestBuild(ctx, channel)
+		if err != nil {
+			return metadata{}, errors.Wrap(err, "failed to fetch latest build")
+		}
 	}
 
 	url := fmt.Sprintf("%s/%s/%s/meta.json", baseURL, channel, build)
diff --git a/pkg/rhcos/qemu.go b/pkg/rhcos/qemu.go
index 8fd30de..c8be19e 100644
--- a/pkg/rhcos/qemu.go
+++ b/pkg/rhcos/qemu.go
@@ -9,7 +9,7 @@ import (
 
 // QEMU fetches the URL of the latest Red Hat CoreOS release.
 func QEMU(ctx context.Context, channel string) (string, error) {
-	meta, err := fetchLatestMetadata(ctx, channel)
+	meta, err := fetchMetadata(ctx, channel, buildName)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to fetch RHCOS metadata")
 	}
```

Reply:
I'm also fine bubbling that `build` argument up into `AMI` and `QEMU`, and dropping the "latest" from their godocs.  But then we'd need to make a public `DefaultBuild` string (defaulting to empty, clobbered during releases) for callers to pass in (like they already pass in `DefaultChannel`).

Reply:
> If we have a hard-coded buildName, I'd rather not bother hitting builds.json at all.

Agree - though I'd note that this obviates the "freshness timestamp" checking you were arguing for and [we implemented](https://github.com/coreos/coreos-assembler/pull/224).  But eh.

Reply:
So I'm still in favor of skipping `builds.json` if we have a compile-time `buildName`, because it's one less network call to go wrong.  If we compile in a name that doesn't exist, I think installer devs can figure it out without [a pretty error message][1].  And by the time this gets handed to users who would not be able to figure out why the value is broken without a list in the error message, I expect there are likely to be a number of other issues (e.g. they may not have a Go compiler, or know where the source is, etc.).

But all of that said, this is your PR ;).  If you're committed to this approach, I'm ok landing it.  Can you change the `'%s'` in this line into `%q`?

[1]: https://github.com/openshift/installer/pull/757#issuecomment-443040143